tests/lean/structInstExtraEta.lean

```diff
@@ -0,0 +1,16 @@
+/-! This test assures that that `toA` field is not eta expanded using the `with` pattern as it was
+  before #2478 -/
+
+structure A where
+
+structure B extends A
+
+def a : A := sorry
+
+def b : B := { a with }
+#print b
+
+structure C extends B
+
+def c : C := { b with }
+#print c
```

Comment:
What I specifically want to see is `def c : C := { a with }`, though there's certainly no harm in having both tests